byway/src/main/webapp/routepage.js

```diff
@@ -11,171 +11,206 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
-
-// copies of recommendations and selected stops, stored as arrays for synchronous updating
-let recs = [];
-let stops = []; // TODO: Make set to avoid duplicates
-
+ 
+// copies of recommendations and selected stops, stored as sets for synchronous updating
+let recs = new Set;
```

Comment:
Missing parenthesis: `let recs = new Set()`. Update this and other set variables.

Reply:
I saw it instantiated as just Set in some other code and didn't question it. My bad, changed it to the proper way.